### DISCUSSION TITLE
Reusing rendered dataframe as table?

### DISCUSSION THREAD

**Discussion by pschumm** [executablebooks/meta#1153] [Q&A] [answered]
Starting to get acquainted with `mystmd` and really loving it! Just one thing that I'm missing. [As shown in the documentation](https://mystmd.org/guide/reuse-jupyter-outputs#outputs-as-figures-or-images), I can generate a figure in a notebook, and then capture that cell's output and include it as a figure in an article with something like:

```
:::{figure} #my-cell
:name: fig-my-cell
:::
```

What I can't seem to figure out is how to do that with a table. I mean, I see that I can use the [table](https://mystmd.org/guide/directives#directive-table) directive with a Markdown table, but suppose my table is a rendered Pandas dataframe, such as that created with [Pandas.crosstab()](https://pandas.pydata.org/docs/reference/api/pandas.crosstab.html)? It's a pretty standard workflow when working in JupyterLab to create tables this way, and it would be great if those could be reused in an article just like figures can be. I'm not asking for any special formatting—just the ability to bring the table over without having to recreate it in Markdown.

Perhaps I'm missing something obvious here? Thanks very much in advance!

**Comment by choldgraf** [comment]
I think you should be able to embed this using the standard embed syntax. So if you have a cell labeled "mytable" that outputs a table, the following two options would embed that table:

```
![](#mytable)
``` 

````
```{embed} #mytable
```
````

**Reply by rowanc1** [reply, marked as answer]
In addition to that you can wrap it in a `table` directive or a figure directive of kind `table`. Which will give you the `source` and the table caption.

![image](https://github.com/executablebooks/meta/assets/913249/ae971e8a-ea91-4111-85d2-855e9c10e288)

```
:::{table} This is my table
:label: tbl:here
![](#pd-table)
:::

:::{figure} #pd-table
:label: tbl:there
:kind: table
This is my table caption
:::
```

Over in jupyter, the cell needs to be labeled with `#| label: pd-table`.

**Reply by choldgraf** [reply]
I've got a docs pr incoming later today or tomorrow when I get some time

**Reply by choldgraf** [reply]
Here's a PR to try and clarify this + some other things:

- https://github.com/executablebooks/mystmd/pull/1069

**Comment by pschumm** [comment]
Thanks very much @choldgraf and @rowanc1! Exactly what I was looking for (just tried and each of the alternatives you provided works like a champ). I really appreciate the timely replies.
